### PR TITLE
added support for CTEs, and potentially DDL return values

### PR DESCRIPTION
**Pull request by dmalan**

### CODE REVIEW

File: tests/sql.py [hunk 1]
```diff
@@ -306,6 +309,9 @@ def test_numeric(self):
         self.assertRaises(RuntimeError, self.db.execute, "INSERT INTO foo VALUES (:1, :2)", 'bar', 'baz', 'qux')
         self.assertRaises(RuntimeError, self.db.execute, "INSERT INTO foo VALUES (:1, :2)", 'bar', baz='baz')
 
+    def test_cte(self):
+        self.assertEqual(self.db.execute("WITH foo AS ( SELECT 1 AS bar ) SELECT bar FROM foo"), [{"bar": 1}])
+
```
Comment (kzidane) [inline tests/sql.py]: Is that a duplicate test?

File: src/cs50/sql.py [hunk 2]
```diff
@@ -120,6 +120,14 @@ def execute(self, sql, *args, **kwargs):
...
             raise RuntimeError("cannot pass both named and positional parameters")
 
+        # Infer command from (unflattened) statement
+        for token in statements[0]:
+            if token.ttype in [sqlparse.tokens.Keyword.DDL, sqlparse.tokens.Keyword.DML]:
+                command = token.value.upper()
+                break
+        else:
+            command = None
+
```
Comment (kzidane) [inline src/cs50/sql.py]: I thinks this is equivalent to the following fwiw:

```
command = next((t.value.upper() for t in statements[0]
            if t.ttype in (sqlparse.tokens.Keyword.DDL, sqlparse.tokens.Keyword.DML)), None)
```

Comment (dmalan) [reply src/cs50/sql.py]: Just not as readable I think!

File: tests/sql.py (hunk 1 above)
Comment (dmalan) [reply tests/sql.py]: MySQL < 8.0 doesn't support CTEs, so I added `test_cte` to the SQLite tests and PostgreSQL tests but not MySQL!

**Pull request merged**